### PR TITLE
Wire up pinTag

### PR DESCRIPTION
**Pull request by inlined**
Adds a new alternative input for functions rewrites that encapsulate options into an object. The new functions rewrite form and run rewrites get a pinTag option. When set, we will resolve or add a new tag to the targeted service and send it in our ServingConfig.

### CODE REVIEW

**Comment by inlined** [comment]
Holy 🤬 git went sideways on this one. Don't worry about this until we can get the other three PRs merged in. Then I'll do whatever is necessary to get a clean diff.

File: src/gcp/proto.ts
Comment (TheIronDev) [inline src/gcp/proto.ts:243]: nit: `This cleans that up`.

File: src/hosting/config.ts
Comment (TheIronDev) [inline src/hosting/config.ts:257]: nit: that

File: src/hosting/config.ts
Comment (TheIronDev) [inline src/hosting/config.ts:258]: I'm guessing because `HostingRewrites` is union typed between a bunch of different things, that while technically legal, you have to give it a strong winky face to get the code to behave (that and we don't know the shape of the config's rewrites when we invoke `normalize()`.

**Review by TheIronDev — APPROVED** [review]
LGTM

**Comment by inlined** [comment]
Thanks for the review @TheIronDev. Since this is Hosting, I'm going to also wait for a review by @bkendall.

File: src/hosting/config.ts
Comment (inlined) [reply src/hosting/config.ts:258]: But `if (typeof rewrite.function === "string")` should indicate that rewrite is a `LegacyFunctionRewrite`, which is where `region` is defined. `rewrite` should have already been typed, but the check against function being a string is only typing that field I think. I may try a different appraoch.

File: src/gcp/proto.ts
Comment (bkendall) [inline src/gcp/proto.ts:245]: Does this have tests?

I'm really confused about what this is solving. The default value of any key that doesn't exist on an object _is_ `undefined`, so when I see below `if (keyable[key] === undefined) delete keyable[key]` I think "well, if a test was testing for the existence of something, this wouldn't have changed anything".  Maybe you can point me to a unit test that is flakey without this function?

File: src/deploy/functions/backend.ts
Comment (inlined) [reply src/deploy/functions/backend.ts:503]: Now that it's grown I could call it LoadFunctionContext. But yes, loadExistingBackend makes existingBackend exist. Added comment.

File: src/deploy/hosting/context.ts
Comment (inlined) [reply src/deploy/hosting/context.ts:6]: Because I added the comment in an earlier PR and in a later PR added a stricter type in response to your comments.

**Pull request merged**
